syntax = "proto3";

//包名
package agent;
//编译输出
//python -m grpc_tools.protoc -I输入包地址 --python_out=输出代码1地址 --grpc_python_out=输出代2地址 .proto文件名

message UserRequest {
  int32 user_id = 1;  // 字段标识号（1开始，唯一，用于序列化）
}

// 2. 定义响应消息类型（服务端向客户端返回的数据）
message UserResponse {
  int32 user_id = 1;
  string username = 2;
  int32 age = 3;
  string email = 4;
}

service UserService {
  // 简单 RPC：定义接口方法（请求类型 -> 响应类型）
  rpc GetUserInfo(UserRequest) returns (UserResponse);
}

